packages/contracts/src/utils/forwarder_wrapper.ts

```diff
@@ -0,0 +1,185 @@
+import { Order, OrderTransactionOpts } from '0x.js';
+import { ContractAbi, TransactionReceiptWithDecodedLogs } from '@0xproject/types';
+import { AbiDecoder, BigNumber } from '@0xproject/utils';
+import { Web3Wrapper } from '@0xproject/web3-wrapper';
+import * as ethUtil from 'ethereumjs-util';
+import * as _ from 'lodash';
+import * as Web3 from 'web3';
+
+import { formatters } from '../../src/utils/formatters';
+import { orderUtils } from '../../src/utils/order_utils';
+import { ForwarderContract } from '../contract_wrappers/generated/forwarder';
+// import { artifacts } from '../ts/artifacts';
+
+import { Artifact, SignatureType, SignedOrder, UnsignedOrder } from './types';
+
+export class ForwarderWrapper {
+    private _web3Wrapper: Web3Wrapper;
+    private _forwarderContract: ForwarderContract;
+    // TODO remove this
+    // public static async getForwarderContractAsync(web3Wrapper: Web3Wrapper): Promise<ForwarderContract> {
+    //     const networkId = await web3Wrapper.getNetworkIdAsync();
+    //     const [abi, address] = await this._getContractAbiAndAddressFromArtifactsAsync(
+    //         artifacts.Forwarder,
+    //         web3Wrapper,
+    //         networkId,
+    //     );
+    //     const contractInstance = new ForwarderContract(web3Wrapper, abi, address);
+    //     return contractInstance;
+    // }
+    protected static async _getContractAbiAndAddressFromArtifactsAsync(
+        artifact: Artifact,
+        web3Wrapper: Web3Wrapper,
+        networkId: number,
+    ): Promise<[ContractAbi, string]> {
+        const contractAddress = this._getContractAddress(artifact, networkId);
+        const doesContractExist = await web3Wrapper.doesContractExistAtAddressAsync(contractAddress);
+        if (!doesContractExist) {
+            throw new Error(`Forwarder Contract ${contractAddress} Not Found on network ${networkId}`);
+        }
+        const abiAndAddress: [ContractAbi, string] = [artifact.networks[networkId].abi, contractAddress];
+        return abiAndAddress;
+    }
+    protected static _getContractAddress(artifact: Artifact, networkId: number, addressIfExists?: string): string {
+        let contractAddress;
+        if (_.isUndefined(addressIfExists)) {
+            contractAddress = artifact.networks[networkId].address;
+            if (_.isUndefined(contractAddress)) {
+                throw new Error('ContractDoesNotExist');
```

Comment:
Our convention is all caps with underscores between words for error messages.

Reply:
This is removed for now with the above TODO remove, we're not using it in tests but we may have a more generic way of initializing these contracts from artifacts.